src/components/AddressResolver/EndpointEntry.tsx

```diff
@@ -0,0 +1,164 @@
+import React, { useState } from "react";
+import styled from "@emotion/styled";
+import { InputDefault } from "../UI/Input";
+import { SvgIcon, SvgIconTrash2, SvgIconSave, SvgIconEdit2 } from "../UI/SvgIcon";
+import { useThirdPartyAPIEndpoints } from "../../common/hooks/useThirdPartyAPIEndpoints";
+import { vars } from "../../styles";
+import { generateUniqueId } from "./../../common/utils/generateUniqueId";
+import isURL from "validator/lib/isURL";
+import isEmpty from "validator/lib/isEmpty";
+
+interface EndpointEntryProps {
+  className?: string;
+  id: string;
+  order: number;
+  removeEndpoint: () => void;
+  api?: string;
+  name?: string;
+  canEdit?: boolean;
+}
+
+export const EndpointEntry = styled(
+  ({ className, id, order, removeEndpoint, api = "", name = "", canEdit = false }: EndpointEntryProps) => {
+    const [isEditable, setEditable] = useState(canEdit);
+    const [inputErrorMessageName, setInputErrorMessageName] = useState("");
+    const [inputMessageEndpoint, setInputErrorMessageEndpoint] = useState("");
+    const [endpointAPI, setEndpointAPIValue] = useState(api);
+    const [endpointName, setEndpointNameValue] = useState(name);
+    const { thirdPartyAPIEndpoints, addThirdPartyAPIEndpoint, setThirdPartyAPIEndpoints } = useThirdPartyAPIEndpoints();
+
+    const onEndpointAPIChanged = (event: React.ChangeEvent<HTMLInputElement>) => {
+      setEndpointAPIValue(event.target.value);
+    };
+
+    const onEndpointNameChanged = (event: React.ChangeEvent<HTMLInputElement>) => {
+      setEndpointNameValue(event.target.value);
+    };
+
+    const editEndpoint = (foundIndex: number) => {
```

Comment:
This is a really funky function that is not immediately understood. See comment below.